### PR TITLE
fixing the timezone in next run id info

### LINKED ISSUE #38172: Timezone information should be shown in Next Run details
### Description

We currently do not show the Timezone information in Next Run information,
although user can hover over and check this  in ToolTip, we can also show this upfront.
image

### Use case/motivation

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Bowrna**
closes: #38172
related: #38172

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/www/templates/airflow/dag.html [hunk 1]
```diff
@@ -151,7 +151,8 @@ <h4 class="pull-right js-dataset-triggered" style="user-select: none;-moz-user-s
         {% endif %}
         {% if dag_model is defined and dag_model.next_dagrun is defined and dag_model.schedule_interval != 'Dataset' %}
           <p class="label label-default js-tooltip" style="margin-left: 5px" id="next-run" data-html="true" data-placement="bottom">
-            Next Run ID: <time datetime="{{ dag_model.next_dagrun }}">{{ dag_model.next_dagrun }}</time>
+            Next Run ID: <time datetime="{{ dag_model.next_dagrun }}">{{ dag_model.next_dagrun }}
+              {{ dag_model.next_dagrun.tzinfo }}</time>
```
Comment (Bowrna) [inline airflow/www/templates/airflow/dag.html:155]: @bbovenzi @pierrejeambrun With the fix added in the dag.html, I could see that time has timezone info (both in text `UTC` and numerical `+00:00` representation ) on loading, but then after loading it goes back to the old state. Am I missing something to notice?

Comment (Bowrna) [inline airflow/www/templates/airflow/dag.html]: @bbovenzi @pierrejeambrun With the fix added in the dag.html, I could see that time has timezone info (both in text UTC and numerical +00:00 representation ) on loading, but then after loading it goes back to the old state. Am I missing something to notice?
Screenshot 2024-03-26 at 12 07 41 PM

Comment (Bowrna) [reply airflow/www/templates/airflow/dag.html]: The above image is during load, and I am attaching below the image which shows up after the page got loaded

Screenshot 2024-03-26 at 12 15 08 PM

Comment (bbovenzi) [reply airflow/www/templates/airflow/dag.html]: The only change needed is to add `data-with-tz="true"` to `<time>`

Comment (Bowrna) [reply airflow/www/templates/airflow/dag.html]: I see @bbovenzi 
I didn't know such an attribute exists for time tag( googling also didn't help). Thank you so much. I have a long way to go in exploring the frontend side.

Comment (bbovenzi) [reply airflow/www/templates/airflow/dag.html]: No worries. I had to poke around the code too. `data-` tags are custom so googling wouldn't have helped.

**Review by bbovenzi — APPROVED** [review]

File: airflow/www/templates/airflow/dag.html (hunk 1 above)
Comment (Bowrna) [reply airflow/www/templates/airflow/dag.html]: I didn't know this. Thanks @bbovenzi I will go and learn more about this

**Pull request merged**